src/Qowaiv/Chemistry/CasRegistryNumber.cs

```diff
@@ -0,0 +1,172 @@
+    [Pure]
+    private string ToXmlString() => ToString(CultureInfo.InvariantCulture);
+
+    /// <summary>Serializes the CAS Registry Number to a JSON node.</summary>
+    /// <returns>
+    /// The serialized JSON string.
+    /// </returns>
+    [Pure]
+    public string? ToJson() => m_Value == default ? null : ToString(CultureInfo.InvariantCulture);
+
+    /// <summary>Deserializes the CAS Registry Number from a JSON number.</summary>
+    /// <param name="json">
+    /// The JSON number to deserialize.
+    /// </param>
+    /// <returns>
+    /// The deserialized CAS Registry Number.
+    /// </returns>
+    [Pure]
+    public static CasRegistryNumber FromJson(long json) => Create(json);
+
+    /// <summary>Casts the CAS Registry Number to a <see cref="long"/>.</summary>
+    public static explicit operator long(CasRegistryNumber val) => val.m_Value;
+
+    /// <summary>Casts a <see cref="long"/> to a CAS Registry Number.</summary>
+    public static explicit operator CasRegistryNumber(long val) => Create(val);
+
+    /// <summary>Converts the <see cref="string"/> to <see cref="CasRegistryNumber"/>.
+    /// A return value indicates whether the conversion succeeded.
+    /// </summary>
+    /// <param name="s">
+    /// A string containing the CAS Registry Number to convert.
+    /// </param>
+    /// <param name="formatProvider">
+    /// The specified format provider.
+    /// </param>
+    /// <param name="result">
+    /// The result of the parsing.
+    /// </param>
+    /// <returns>
+    /// True if the string was converted successfully, otherwise false.
+    /// </returns>
+    [Pure]
+    public static bool TryParse(string? s, IFormatProvider? formatProvider, out CasRegistryNumber result)
+    {
+        result = default;
+        var str = s.Unify();
+        if (string.IsNullOrEmpty(str))
+        {
+            return true;
+        }
+        else if (Qowaiv.Unknown.IsUnknown(str, formatProvider as CultureInfo))
+        {
+            result = Unknown;
+            return true;
+        }
+        else return long.TryParse(str, out var number) && TryCreate(number, out result);
+    }
+
+    /// <summary>Creates a CAS Registry Number from a long.
+    /// A return value indicates whether the creation succeeded.
+    /// </summary>
+    /// <param name="val">
+    /// A long describing a CAS Registry Number.
+    /// </param>
+    /// <param name="result">
+    /// The result of the creation.
+    /// </param>
+    /// <returns>
+    /// True if a CAS Registry Number was created successfully, otherwise false.
+    /// </returns>
+    public static bool TryCreate(long val, out CasRegistryNumber result)
+    {
+        result = default;
+        if (val == 0) return true;
+        else if (val >= 10_000 && val <= 9_999_999_999)
+        {
+            var checksum = val % 10;
+            var buffer = val / 10;
+            var sum = 0L;
+            var factor = 0;
+
+            while (buffer > 0)
```

Comment:
Je doet buffer alleen maar delen door 10 toch? Dan wordt het nooit kleiner dan nul?

Reply:
It's a long, so it stays a long. ;)